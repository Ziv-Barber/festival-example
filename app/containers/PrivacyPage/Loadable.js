import loadable from 'loadable-components'

/**
 * Asynchronously loads the PrivacyPage component.
 */
export default loadable(() => import('./index'))
