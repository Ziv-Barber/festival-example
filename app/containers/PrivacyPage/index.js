// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { withRouter } from 'react-router'

// Other components that we need:
import PageLogo from '../../styled/PageLogo'
import PrivactText from '../../styled/PrivacyText'
import Footer from '../../styled/Footer'

import infoContent from '../../content/info'

import pageBack from '../../images/privacy.jpg'

/*
 * PrivacyPage
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */
/* eslint-disable react/prefer-stateless-function */
class PrivacyPage extends React.PureComponent<{
  location: { pathname: string }
}> {
  componentDidUpdate(prevProps) {
    if (this.props.location.pathname !== prevProps.location.pathname) {
      window.scrollTo(0, 0)
    }
  }

  render() {
    return (
      <React.Fragment>
        <PageLogo back={pageBack}>Privacy</PageLogo>
        <PrivactText>{infoContent.privacy}</PrivactText>
        <Footer />
      </React.Fragment>
    )
  }
}

// $FlowFixMe
export default withRouter(PrivacyPage)
