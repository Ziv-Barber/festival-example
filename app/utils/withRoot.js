// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
