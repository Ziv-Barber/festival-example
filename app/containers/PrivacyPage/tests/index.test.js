// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'
import { BrowserRouter as Router } from 'react-router-dom'

import PrivacyPage from '../index'
import withRoot from '../../../utils/withRoot'
import { IntlProvider } from 'react-intl'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {PrivacyPage} */
describe('<PrivacyPage />', () => {
  afterEach(cleanup)

  /** @test {PrivacyPage} */
  it('should render', () => {
    const TestComp = (prop) => (
      <IntlProvider locale="en">
        <Router>
          <PrivacyPage />
        </Router>
      </IntlProvider>
    )

    const App = withRoot(TestComp)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
