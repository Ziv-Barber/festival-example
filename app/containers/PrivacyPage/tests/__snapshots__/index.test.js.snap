// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivacyPage /> should render 1`] = `
<div>
  <div
    class="MuiPaper-root-1 MuiPaper-elevation0-3 PageLogo-sc-14ncvse-0 idXUci"
  >
    <h2
      class="MuiTypography-root-28 MuiTypography-h5-44 pageLogoText"
    >
      Privacy
    </h2>
  </div>
  <h2
    class="MuiTypography-root-28 MuiTypography-h5-44 PrivacyText-rns7yr-0 hRsTIG"
  >
    The operators of this website take the protection of your personal data very seriously. We treat your personal data as confidential and in accordance with the statutory data protection regulations and this privacy policy. If you use this website, various pieces of personal data will be collected. Personal information is any data with which you could be personally identified. This privacy policy explains what information we collect and what we use it for. It also explains how and for what purpose this happens. Please note that data transmitted via the internet (e.g. via email communication) may be subject to security breaches. Complete protection of your data from third-party access is not possible.
  </h2>
  <div
    class="MuiPaper-root-1 MuiPaper-elevation0-3 Footer-sc-15rsv6x-0 kQiiTI"
  >
    <div
      class="MuiGrid-container-64 MuiGrid-spacing-xs-8-86 MuiGrid-align-items-xs-center-72 MuiGrid-justify-xs-center-81 listGrid"
    >
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-2-94 MuiGrid-grid-sm-2-108 MuiGrid-grid-md-2-122 MuiGrid-grid-lg-1-135 MuiGrid-grid-xl-1-149 listItem"
      >
        <a
          class="link"
          href="/newsletter"
        >
          <button
            class="MuiButtonBase-root-187 MuiButton-root-161 MuiButton-text-163 MuiButton-flat-166"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-162"
            >
              Newsletter
            </span>
            <span
              class="MuiTouchRipple-root-190"
            />
          </button>
        </a>
      </div>
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-2-94 MuiGrid-grid-sm-2-108 MuiGrid-grid-md-2-122 MuiGrid-grid-lg-1-135 MuiGrid-grid-xl-1-149 listItem"
      >
        <a
          class="link"
          href="/press"
        >
          <button
            class="MuiButtonBase-root-187 MuiButton-root-161 MuiButton-text-163 MuiButton-flat-166"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-162"
            >
              Press
            </span>
            <span
              class="MuiTouchRipple-root-190"
            />
          </button>
        </a>
      </div>
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-2-94 MuiGrid-grid-sm-2-108 MuiGrid-grid-md-2-122 MuiGrid-grid-lg-1-135 MuiGrid-grid-xl-1-149 listItem"
      >
        <a
          class="link"
          href="/privacy"
        >
          <button
            class="MuiButtonBase-root-187 MuiButton-root-161 MuiButton-text-163 MuiButton-flat-166"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-162"
            >
              Privacy
            </span>
            <span
              class="MuiTouchRipple-root-190"
            />
          </button>
        </a>
      </div>
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-2-94 MuiGrid-grid-sm-2-108 MuiGrid-grid-md-2-122 MuiGrid-grid-lg-1-135 MuiGrid-grid-xl-1-149 listItem"
      >
        <a
          class="link"
          href="/contact"
        >
          <button
            class="MuiButtonBase-root-187 MuiButton-root-161 MuiButton-text-163 MuiButton-flat-166"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-162"
            >
              Contact
            </span>
            <span
              class="MuiTouchRipple-root-190"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
